### ISSUE TITLE
Suggestion: Make Object3D.copy non-recursive by default

### ISSUE THREAD

**Issue by speigg** [mrdoob/three.js#16739]
I often find myself accidentally causing memory to grow exponentially between frames (or an infinite loop within a single frame), by assuming that `Object3D.copy` is non-recursive by default. Am I the only one who does this? I think making it non-recursive by default, requiring `true` to be passed for a recursive copy, might be more intuitive.

**Comment by WestLangley** [comment]
I am curious as to your use case. What are you copying and how frequently? How are you generating infinite loops?

**Comment by speigg** [comment]
I suppose I expect it to work as a shorthand for copying position/quarternion/scale, and any other class-specific properties like fov, aspectRatio, and projectionMatrix, etc.  

For example, I’ll have `camera.copy(otherCamera)` in the update loop, with the intention to copy the camera settings. However, if that other camera instance had children, they will be copied and added each frame, increasing the memory each frame. If copying properties from a parent object, that causes the infinite loop.

**Comment by WestLangley** [comment]
>If copying properties from a parent object, that causes the infinite loop.

Can you please describe your specific example?

**Comment by WestLangley** [comment]
>if that other camera instance had children, they will be copied and added each frame

Well, this is bad. In `Object3D.copy()`:
```js
if ( recursive === true ) {

	for ( var i = 0; i < source.children.length; i ++ ) {

		var child = source.children[ i ];
		this.add( child.clone() );

	}

}
```

Every call to `copy()` adds more children -- and clones them no less.

**Comment by speigg** [comment]
The infinite loop was the side effect of a logical error, not a use case; I started with a child camera attached to a parent camera (so that they would move together), and at one point I added effectively `childCamera.copy(parentCamera)` in order to copy the other camera properties over and readjust them. Bad idea for a few reasons (I should have just attached the second camera to the first camera’s parent), but since the child camera is itself cloned and added a child of itself it resulted in an infinite loop. The real trickiness is in the other use case, however, not this one. 

For example, if copying from one camera to another camera each frame, everything works fine as long as the copied camera has no children. As soon as a child is added to that camera (e.g., throughout the course of the app running), the memory starts growing quickly as that child object is repeatedly cloned, and eventually crashes the page as it runs out of memory. The last time this happened it took me quite a while to find that memory leak and trace it back to `Object3D.copy`. I’ll probably remember now, as after having this happen to me twice it’s hard to forget; nevertheless, it might save other people from some frustration if the copy was non-recursive by default.

**Comment by WestLangley** [comment]
>The infinite loop was the side effect of a logical error, not a use case;

OK, forget that.

> copying from one camera to another camera each frame

Can you assign, not copy?

```js
var camera1 = new THREE.PerspectiveCamera( ... );
camera1.add( light1 );

var camera2 = new THREE.OrthographicCamera( ... );
camera2.add( light2 );

...

currentCamera = camera1;
```

**Comment by speigg** [comment]
> Can you assign, not copy?

no, because I don't want to modify the original. The pattern is to copy the settings of another object, and then modify them in some way (e.g., changing the fov while keeping the aspect ratio and position/rotation/scale). Something like:

```
camera.copy(otherCamera)
camera.fov = 30
```

**Comment by mrdoob** [comment]
Changing that default would be a good way of breaking a lot of projects 😈
Maybe `camera.copy( otherCamera, false )` could be an option though.
